From 22a8ab46ccb128f93557b18b9325b266829b68ad Mon Sep 17 00:00:00 2001
From: Nikolai Kondrashov <Nikolai.Kondrashov@redhat.com>
Date: Mon, 10 Sep 2012 14:11:40 +0300
Subject: [PATCH] Add uninstall command hints to ipa-*-install

Add uninstall command to the uninstall instructions in the "already
installed" responses of ipa-server-install, ipa-client-install and
ipa-replica-install.

https://fedorahosted.org/freeipa/ticket/3065
---
 install/tools/ipa-replica-install         |    5 +++--
 install/tools/ipa-server-install          |   10 ++++++----
 ipa-client/ipa-install/ipa-client-install |    3 ++-
 3 files changed, 11 insertions(+), 7 deletions(-)

diff --git a/install/tools/ipa-replica-install b/install/tools/ipa-replica-install
index 92e5f8659d3d96f1b37540f5b7f17ea7d869a6d2..b56fa2ea5826b6fe25e6db02c7e640e50bca0790 100755
--- a/install/tools/ipa-replica-install
+++ b/install/tools/ipa-replica-install
@@ -407,8 +407,9 @@ def main():
 
     client_fstore = sysrestore.FileStore('/var/lib/ipa-client/sysrestore')
     if client_fstore.has_files():
-        sys.exit("IPA client is already configured on this system.\n"
-                 + "Please uninstall it first before configuring the replica.")
+        sys.exit("IPA client is already configured on this system.\n" +
+            "Please uninstall it first before configuring the replica, " +
+            "using 'ipa-client-install --uninstall'.")
 
     global sstore
     sstore = sysrestore.StateFile('/var/lib/ipa/sysrestore')
diff --git a/install/tools/ipa-server-install b/install/tools/ipa-server-install
index 37d37b81a25c391e6b029b5918d440846a617210..33e1c257fe4b1d19e98c548875c46707e097d2c2 100755
--- a/install/tools/ipa-server-install
+++ b/install/tools/ipa-server-install
@@ -574,14 +574,16 @@ def main():
         print "\nThe log file for this installation can be found in /var/log/ipaserver-install.log"
         if not options.external_ca and not options.external_cert_file and is_ipa_configured():
             installation_cleanup = False
-            sys.exit("IPA server is already configured on this system.\n"
-                     + "If you want to reinstall the IPA server please uninstall it first.")
+            sys.exit("IPA server is already configured on this system.\n" +
+                "If you want to reinstall the IPA server, please uninstall " +
+                "it first using 'ipa-server-install --uninstall'.")
 
         client_fstore = sysrestore.FileStore('/var/lib/ipa-client/sysrestore')
         if client_fstore.has_files():
             installation_cleanup = False
-            sys.exit("IPA client is already configured on this system.\n"
-                     + "Please uninstall it first before configuring the IPA server.")
+            sys.exit("IPA client is already configured on this system.\n" +
+                "Please uninstall it before configuring the IPA server, " +
+                "using 'ipa-client-install --uninstall'")
 
     root_logger.debug('%s was invoked with options: %s' % (sys.argv[0], safe_options))
     root_logger.debug("missing options might be asked for interactively later\n")
diff --git a/ipa-client/ipa-install/ipa-client-install b/ipa-client/ipa-install/ipa-client-install
index 8181e5a19d9e3be4607aa7aa57e4ddfea9b49bf9..190efb183d8c96e2c9665cf51d5346dc1111ae24 100755
--- a/ipa-client/ipa-install/ipa-client-install
+++ b/ipa-client/ipa-install/ipa-client-install
@@ -1921,7 +1921,8 @@ def main():
     if fstore.has_files():
         root_logger.error("IPA client is already configured on this system.")
         root_logger.info(
-            "If you want to reinstall the IPA client, uninstall it first.")
+            "If you want to reinstall the IPA client, uninstall it first " +
+            "using 'ipa-client-install --uninstall'.")
         return CLIENT_ALREADY_CONFIGURED
 
     rval = install(options, env, fstore, statestore)
-- 
1.6.0.6

